Cargo.toml

```diff
@@ -19,6 +19,7 @@ json = ["regex"]
 [dependencies]
 regex = {version = "1", optional = true }
 futures = {version = "0.3.5", optional = true }
+itertools = "0.10.0"
```

Comment:
Have you benched performance over this? parsing a simple string how it affects? Just need to run `cargo bench parse` for master and for this branch. I would be really interested to know this on itertools.

Reply:
I looked at it, and it turns out that `itertools::fold_while` is only a thin wrapper around the stdlib's `try_fold`, so I just changed the code to use that. And it actually does improve the performance a lot to just use `try_fold`.